discovery-infra/test_infra/controllers/node_controllers/node_controller.py

```diff
@@ -12,7 +12,9 @@
 
 class NodeController(ABC):
 
-    def __init__(self, config: BaseNodeConfig, cluster_config: BaseClusterConfig):
+    T = TypeVar('T', bound=SupportsAbs[BaseNodeConfig])
```

Comment:
Annotations supposed to make thing easier to read.
And why not just `BaseNodeConfig`?

Reply:
It's a generic type with constraint that the config object should be a subtype of BaseNodeConfig
This way the compiler knows the real type in the derived controller class like VSphereController